containers/anndata-to-ui/context/main.py

```diff
@@ -44,14 +47,27 @@ def main(input_dir, output_dir):
         for layer in adata.layers:
             if isinstance(adata.X, sparse.spmatrix):
                 adata.X = adata.X.tocsc()
+    
         # All data from secondary_analysis is scaled at the moment to zero-mean unit-variance
         # https://github.com/hubmapconsortium/salmon-rnaseq/blob/master/bin/analysis/scanpy_entry_point.py#L47
         # We currently cannot visaulize this in Vitessce so we replace `X` with the raw counts.
-        if 'spliced_unspliced_sum' in adata.layers:
-            adata.layers['scaled'] = adata.X
-            adata.X = adata.layers['spliced_unspliced_sum']
+        # Ideally, we should be able to manage the `layers` and `X` simultaneously in `zarr` but currently we cannot:
+        # https://github.com/theislab/anndata/issues/524
+        if (SECONDARY_ANALYSIS == h5ad_file):
+            # Need .copy() to prevent scanpy from altering the original layer.
+            adata.layers['scaled'] = adata.X.copy()
+            adata.X = adata.layers[assay.secondary_analysis_layer].copy()
```

Comment:
This can now always be `adata.layers['unscaled']` across assays, with the new `salmon-rnaseq` release.

Reply:
velocity analysis included?

Reply:
(I don't think so judging by your recent commit but just making sure).